### PR TITLE
Added the ability for microbenchmarks to print the probability of the result.

### LINKED ISSUE #101147: Add statistics to microbenchmarks results
Right now microbenchmarks just print out the mean value.  They should also include standard deviation and/or confidence less than as a way to measure how good the benchmark is.

### PR DESCRIPTION
**Pull request by gaaclarke**
fixes https://github.com/flutter/flutter/issues/101147
related https://github.com/flutter/flutter/pull/100926

Probability is measured as the probability that the mean is +-5% of the actual value.

Test exempt: it is a test.

## Pre-launch Checklist

- [x] I read the [Contributor Guide] and followed the process outlined there for submitting PRs.
- [x] I read the [Tree Hygiene] wiki page, which explains my responsibilities.
- [x] I read and followed the [Flutter Style Guide], including [Features we expect every widget to implement].
- [x] I signed the [CLA].
- [x] I listed at least one issue that this PR fixes in the description above.
- [x] I updated/added relevant documentation (doc comments with `///`).
- [ ] I added new tests to check the change I am making, or this PR is [test-exempt].
- [x] All existing and new tests are passing.

If you need help, consider asking for advice on the #hackers-new channel on [Discord].

[Contributor Guide]: https://github.com/flutter/flutter/wiki/Tree-hygiene#overview
[Tree Hygiene]: https://github.com/flutter/flutter/wiki/Tree-hygiene
[test-exempt]: https://github.com/flutter/flutter/wiki/Tree-hygiene#tests
[Flutter Style Guide]: https://github.com/flutter/flutter/wiki/Style-guide-for-Flutter-repo
[Features we expect every widget to implement]: https://github.com/flutter/flutter/wiki/Style-guide-for-Flutter-repo#features-we-expect-every-widget-to-implement
[CLA]: https://cla.developers.google.com/
[flutter/tests]: https://github.com/flutter/tests
[breaking change policy]: https://github.com/flutter/flutter/wiki/Tree-hygiene#handling-breaking-changes
[Discord]: https://github.com/flutter/flutter/wiki/Chat

### CODE REVIEW

File: dev/benchmarks/microbenchmarks/lib/common.dart [hunk 1]
```diff
@@ -33,6 +77,28 @@ class BenchmarkResultPrinter {
     _results.add(_BenchmarkResult(description, value, unit, name));
   }
 
+  /// Adds a benchmark result to the list of results and a probability of that
+  /// result.
+  ///
+  /// The probably is calculated as the probability that the mean is +- 5% of
```
Comment (dnfield) [inline dev/benchmarks/microbenchmarks/lib/common.dart:83]:
```suggestion
  /// The probability is calculated as the probability that the mean is +- 5% of
```

File: dev/benchmarks/microbenchmarks/lib/common.dart [hunk 2]
```diff
@@ -33,6 +77,28 @@ class BenchmarkResultPrinter {
...
+    required String unit,
+    required String name,
+  }) {
+    final double mean = _doMean(values);
+    final double stddev = _doStddev(values, mean);
+    const double margin = 0.05;
+    final double probability = _doProbability(mean: mean, stddev: stddev, margin: margin);
+    _results.add(_BenchmarkResult(description, mean, unit, name));
+    _results.add(_BenchmarkResult('$description - probability margin of error $margin', probability,
+        'percent', '${name}_probability'));
```
Comment (dnfield) [inline dev/benchmarks/microbenchmarks/lib/common.dart:99]: Consider adding the median here

**Review by dnfield — APPROVED** [review]
This looks cool, it'd probably be good to add the median if we're not already.

File: dev/benchmarks/microbenchmarks/lib/common.dart (hunk 2 above)
Comment (dnfield) [reply dev/benchmarks/microbenchmarks/lib/common.dart:99]: And maybe the min/max?

Comment (gaaclarke) [reply dev/benchmarks/microbenchmarks/lib/common.dart:99]: Those values are going to be really noisy so I don't know if we want them showing up in skia perf graphs.  The probability of %5 error is something that should be stable and if it increases it's a regression.  It means our benchmarks have become less reliable for some reason.

**Pull request merged**